src/DateTimePicker.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import cx from 'classnames';
+import Flatpickr from './FlatpickrComponent';
+import TimeInput from './TimeInput';
+import DateTimeLocalInput from './DateTimeLocalInput';
+
+/**
+ * @module DateTimePicker
+ */
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		super(props);
+
+		const propsValue = props.value || {};
+		this.state = {
+			value: {
+				date: propsValue.date || '',
+				time: propsValue.time || ''
+			},
+			isDateTimeLocalSupported: false
+		};
+
+		this.setDate = this.setDate.bind(this);
+		this.setTime = this.setTime.bind(this);
+		this.parseDateTime = this.parseDateTime.bind(this);
+	}
+
+	componentWillMount() {
+		this.setState({ isDateTimeLocalSupported: this.hasBrowserSupport() });
+	}
+
+	hasBrowserSupport() {
+		const input = document.createElement('input'),
+			invalidValue = 'notadate';
+
+		// some browsers (as Android stock browsers) pretend they support
+		// certain input types,
+		// so we set the value to an invalid value and see if browser rejects
+
+		input.setAttribute('type', 'datetime-local');
+		input.setAttribute('value', invalidValue);
+
+		return !(this.props.forceFlatpickr || input.value === invalidValue);
+	}
+
+	setDate(value) {
+		this.setState({ value: { date: value }});
+	}
+
+	setTime(value) {
+		this.setState({ value: { time: value }});
+	}
+
+	parseDateTime(value) {
+		const datetime = value.split('T'),
```

Comment:
this is probably fine, but it appears fragile. Any way we can use the `Date` API to work with this? Something like

```js
const datetime = new Date(value);
const newState = {
  date: datetime.getDate();
  time: `${datetime.getHours()}:${datetime.getMinutes()}`
};
```

or something - maybe that's overkill. Alternatively, you should document the expected sting format of `value` here so that it's more clear what is happening

---

After thinking about it a bit more, the parsing + getting of time strings is a very manual process that should ideally be handled entirely by JavaScript Date objects - the less you have to manipulate time string values, the better.

To do that, you can look into keeping a real `Date` object in state, and then getting/setting time and date info from a single coherent object rather than trying to combine and separate separate `date` and `time` strings.

This function would then be as simple as

```js
parseDateTime(value) {
  this.setState({ datetime: new Date(value) });
}
```

`getDateTime` would be

```js
getDateTime() { return this.state.datetime; }
```

`setDate` and `setTime` would be slightly more complicated, but in a good way IMO:

```js
setDate(value) {
  const newDate = new Date(value);
  const datetime = new Date(this.state.datetime);  // copy current state
  datetime.setFullYear(  // update only the date part
    newDate.getFullYear(),
    newDate.getMonth(),
    newDate.getDate()
  );
  this.setState({ datetime });  // apply setting
},
setTime(value) {
  const newTime = new Date(value);
  const datetime = new Date(this.state.datetime);  // copy current state
  datetime.setHours(  // update only the time part
    newDate.getHours(),
    newDate.getMinutes(),
  );
  this.setState({ datetime });  // apply setting
}
```